samwise.js

```diff
@@ -1,16 +1,15 @@
-/**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.